### ISSUE TITLE
CMake Error: Couldn't compile thirdparty/boost with current configuration [7996]

### ISSUE THREAD

**Issue by nbaldy** [eProsima/Fast-RTPS#1076]
On compilation with current master (e4907e61325cec47419424a4ac5ce21c96bd51ab) 3rdParty/Boost errors happen when attempting to run "colcon build" or "cmake .." (tried with both manual and  colcon installation)

Note that 1e403b47d9bfe4dff3455d55ba7638c09919e4f2 builds with warnings (and is what I will be using) - I just wanted to point out this issue. Sorry if I missed a step, I'm still very new at everything

## Expected Behavior
Successful compilation

## Current Behavior
```
The following error chain:
-- Setting build type to 'Release' as none was specified.
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring Fast RTPS
-- Version: 1.10.0
-- To change the version modify the file configure.ac
-- Performing Test SUPPORTS_CXX14
-- Performing Test SUPPORTS_CXX14 - Success
-- Performing Test SUPPORTS_CXX1Y
-- Performing Test SUPPORTS_CXX1Y - Success
-- Performing Test SUPPORTS_CXX11
-- Performing Test SUPPORTS_CXX11 - Success
-- Check if the system is big endian
-- Searching 16 bit integer
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- fastcdr library found...
-- Found tinyxml2: /usr/lib/x86_64-linux-gnu/libtinyxml2.so  
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_kill
-- Looking for pthread_kill - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
cmakCMake Error at cmake/modules/FindThirdpartyBoost.cmake:31 (message):
  Couldn't compile thirdparty/boost with current configuration!!!-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
cmakCMake Error at cmake/modules/FindThirdpartyBoost.cmake:31 (message):
  Couldn't compile thirdparty/boost with current configuration!!!

  Change Dir:
  /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/build/CMakeFiles/CMakeTmp
  Run Build Command(s):/usr/bin/make cmTC_c924e/fast && /usr/bin/make -f
  CMakeFiles/cmTC_c924e.dir/build.make CMakeFiles/cmTC_c924e.dir/build
  make[1]: Entering directory
 '/home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/build/CMakeFiles/CMakeTmp'
  Building CXX object
  CMakeFiles/cmTC_c924e.dir/ThirdpartyBoostCompile_test.cpp.o
  /usr/bin/c++
  -I/home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include
  -Wall -pedantic -Wextra -Wno-unknown-pragmas
  -Wno-error=deprecated-declarations -o
  CMakeFiles/cmTC_c924e.dir/ThirdpartyBoostCompile_test.cpp.o -c
  /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/test/ThirdpartyBoostCompile_test.cpp

  In file included from
 /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/intrusive/bstree_algorithms.hpp:22:0,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/intrusive/rbtree_algorithms.hpp:33,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/intrusive/detail/rbtree_node.hpp:28,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/intrusive/set_hook.hpp:20,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/intrusive/rbtree.hpp:21,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/intrusive/set.hpp:20,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/indexes/iset_index.hpp:30,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/segment_manager.hpp:36,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/detail/managed_memory_impl.hpp:30,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/managed_shared_memory.hpp:25,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/test/ThirdpartyBoostCompile_test.cpp:16:


  /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/intrusive/detail/math.hpp:276:34:
  warning: use of C++11 long long integer constant [-Wlong-long]

      static const SizeType value = 0xb504f333f9de6484ull
                                    ^
  In file included from
  /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/smart_ptr/shared_ptr.hpp:28:0,

                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/shared_ptr.hpp:17,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/date_time/microsec_time_clock.hpp:18,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/detail/posix_time_types_wrk.hpp:31,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/sync/scoped_lock.hpp:35,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/allocators/detail/allocator_common.hpp:32,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/allocators/allocator.hpp:30,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/segment_manager.hpp:38,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/detail/managed_memory_impl.hpp:30,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/interprocess/managed_shared_memory.hpp:25,
                   from /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/test/ThirdpartyBoostCompile_test.cpp:16:
 /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/smart_ptr/detail/shared_count.hpp:
  In member function ‘void*
  boost::detail::shared_count::get_local_deleter(const sp_typeinfo&)
  const’:

  /home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/thirdparty/boost/include/boost/smart_ptr/detail/shared_count.hpp:508:26:
  error: ‘class boost::detail::sp_counted_base’ has no member named
  ‘get_local_deleter’

           return pi_? pi_->get_local_deleter( ti ): 0
                            ^
  CMakeFiles/cmTC_c924e.dir/build.make:65: recipe for target
  'CMakeFiles/cmTC_c924e.dir/ThirdpartyBoostCompile_test.cpp.o' failed

  make[1]: *** [CMakeFiles/cmTC_c924e.dir/ThirdpartyBoostCompile_test.cpp.o]
  Error 1

  make[1]: Leaving directory
  '/home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/build/CMakeFiles/CMakeTmp'

  Makefile:121: recipe for target 'cmTC_c924e/fast' failed

  make: *** [cmTC_c924e/fast] Error 2

Call Stack (most recent call first):
  CMakeLists.txt:223 (find_package)

-- Configuring incomplete, errors occurred!
See also "/home/local/RESQUARED/nicole.baldy/c01_simulator/3rdParty/src/fastrtps/build/CMakeFiles/CMakeOutput.log".

```

## Steps to Reproduce
Manual Installation or Colcon installation seem to result in the same error message, though that particular one was copied from the manual installation steps. Because I tried the Colcon installation first, the "manual installation" simply used all of the git repos pulled in by the colcon build attempt, like such:
```
wget https://raw.githubusercontent.com/eProsima/Fast-RTPS/master/fastrtps.repos
mkdir src
vcs import src < fastrtps.repos
# Fast CDR
mkdir Fast-CDR/build && cd Fast-CDR/build
cmake ..
cmake --build . --target install
# Foonathan_memory_vendor
cd ../../foonathan_memory_vendor
mkdir build && cd build
cmake ..
cmake --build . --target install
# FastRTPS 
cd ../..
mkdir Fast-RTPS/build && cd Fast-RTPS/build
cmake ..
```
`cmake ..` failed

## System information

- Fast-RTPS version: e4907e61325cec47419424a4ac5ce21c96bd51ab (current Master - "Import Boost Processes") 
- OS: Ubuntu 16.04
- Network interfaces: N/A (I think)
- ROS2: N/A 

## Additional context
Noted in the description, but commit 1e403b47d9bfe4dff3455d55ba7638c09919e4f2 builds for me with some warnings! 
I just wanted to point out the bug in case someone else runs into it!

## Additional resources
- Wireshark capture N/A (I think)
- XML profiles file N/A (I think)

**Comment by MiguelCompany** [comment]
@nbaldy Can you try with -DTHIRDPARTY=ON ?

**Comment by nbaldy** [comment]
`cmake .. -DTHIRDPARTY=ON` failed as well, looks like it is the same error

**Comment by MiguelCompany** [comment]
@nbaldy This should have been fixed by #1078

**Comment by nbaldy** [comment]
Yup, it built! Thanks!
